chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -9506,6 +9507,431 @@ func TestCCIPRouter(t *testing.T) {
 				})
 			})
 
+			sendChunk := func(
+				t *testing.T,
+				i int,
+				chunkSize int,
+				numChunks int,
+				rawReport []byte,
+				buffer_id [32]byte,
+				bufferPDA solana.PublicKey,
+				transmitter solana.PrivateKey,
+			) *ccip_offramp.Instruction {
+				start := i * chunkSize
+				end := min(((i + 1) * chunkSize), len(rawReport))
+
+				chunk := rawReport[start:end]
+
+				appendIx, appendErr := ccip_offramp.NewBufferExecutionReportInstruction(
+					buffer_id[:],
+					uint32(len(rawReport)),
+					chunk,
+					uint8(i),
+					uint8(numChunks),
+					bufferPDA,
+					config.OfframpConfigPDA,
+					transmitter.PublicKey(),
+					solana.SystemProgramID).ValidateAndBuild()
+
+				require.NoError(t, appendErr)
+
+				return appendIx
+			}
+
+			t.Run("Execution report pre-buffering", func(t *testing.T) {
+				getLamports := func(account solana.PublicKey) uint64 {
+					out, err := solanaGoClient.GetBalance(ctx, account, rpc.CommitmentConfirmed)
+					require.NoError(t, err)
+					return out.Value
+				}
+
+				transmitter := getTransmitter()
+				initialLamports := getLamports(transmitter.PublicKey())
+				bufferID := [32]byte{}
+				bufferPDA, _, _ := solana.FindProgramAddress([][]byte{[]byte("execution_report_buffer"), bufferID[:], transmitter.PublicKey().Bytes()}, config.CcipOfframpProgram)
+				// Deliberately chosen not a whole divider, to prove that a smaller terminator chunk is accepted.
+				const chunkSize = 300
+				const terminatorSize = 100
+				data := make([]byte, 1000)
+				for i := range data {
+					data[i] = byte(i % 256)
+				}
+
+				totalSolSpentInFees := uint64(0)
+
+				ix := sendChunk(t, 0, 0, 4, data, bufferID, bufferPDA, transmitter)
+				result := testutils.SendAndFailWith(ctx, t, solanaGoClient, []solana.Instruction{ix}, transmitter, config.DefaultCommitment, []string{"Error Code: ExecutionReportBufferInvalidChunkSize"})
+				totalSolSpentInFees += result.Meta.Fee
+				ix = sendChunk(t, 3, terminatorSize, 4, data, bufferID, bufferPDA, transmitter)
+				result = testutils.SendAndConfirm(ctx, t, solanaGoClient, []solana.Instruction{ix}, transmitter, config.DefaultCommitment)
+				totalSolSpentInFees += result.Meta.Fee
+				ix = sendChunk(t, 3, terminatorSize, 4, data, bufferID, bufferPDA, transmitter)
+				result = testutils.SendAndFailWith(ctx, t, solanaGoClient, []solana.Instruction{ix}, transmitter, config.DefaultCommitment, []string{"Error Code: ExecutionReportBufferAlreadyContainsChunk"})
+				totalSolSpentInFees += result.Meta.Fee
+				ix = sendChunk(t, 0, chunkSize, 4, data, bufferID, bufferPDA, transmitter)
+				result = testutils.SendAndConfirm(ctx, t, solanaGoClient, []solana.Instruction{ix}, transmitter, config.DefaultCommitment)
+				totalSolSpentInFees += result.Meta.Fee
+				// Checking again after the terminator shuffling
```

Comment:
We no longer have terminator shuffling, right? Could the testcase be simplified?

Reply:
I think I'll remove the comment, but there's still value in keeping this somewhat convoluted order to catch edge cases :)